src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -201,7 +201,8 @@ public boolean isNameVisibleToUser(
         // Early return if user is giver
         if (question.giverType == FeedbackParticipantType.TEAMS) {
             // if response is given by team, then anyone in the team can see the response
-            if (roster.isStudentsInSameTeam(response.giver, userEmail)) {
+            if (roster.isStudentInTeam(userEmail, response.giver)
+                    || roster.isStudentsInSameTeam(userEmail, response.giver)) {
```

Comment:
Remember to add comment explain the legacy case

Reply:
If it is not too difficult, we can have a test for this

Reply:
How would I go about writing tests for this exactly? I had thought it was not necessary since the current tests already check for visibility.

Reply:
The newly added condition is not covered right?